Documentation/nnstreamer_capi.md

```diff
@@ -0,0 +1,318 @@
+  int status = ml_pipeline_construct (pipeline, NULL, NULL, &handle);
+```
+
+`valve1` is the name of the valve of the pipeline, and it is opened as a default. This valve can be controlled with getting `ml_pipeline_valve_h`.
+
+```bash
+  ml_pipeline_h handle;
+  ml_pipeline_valve_h valve1;
+
+  status = ml_pipeline_valve_get_handle (handle, "valve1", &valve1);
+```
+
+After start pipeline, the stream of a pipeline can be controlled with valve.
+```bash
+  status = ml_pipeline_start (handle);
+
+  status = ml_pipeline_valve_set_open (valve1, true); /* close */
+```
+
+It can be opened with the same way.
+
+```bash
+  status = ml_pipeline_valve_set_open (valve1, true); /* open */
+```
+
+Before destroy the pipeline, `ml_pipeline_valve_h` has to be released.
+
+```bash
+  status = ml_pipeline_valve_release_handle (valve1); /* release valve handle */
+```
+
+* #### Switch <a name="switchelement"></a>
+When a pipeline consists of multiple branch, and only one of the branches has to be working, this element is useful.
+
+![input-selector](./media/input-selector.png)
+
+```bash
+  gchar * pipeline = g_strdup ("input-selector name=ins ! tensor_converter ! tensor_sink name=sinkx "
+      "videotestsrc is-live=true ! videoconvert ! ins.sink_0 "
+      "videotestsrc num-buffers=3 is-live=true ! videoconvert ! ins.sink_1");
+
+```
+
+First, getting `ml_pipeline_switch_h` is required. The name of switch in this pipeline is `ins`
+
+```bash
+  ml_pipeline_h handle;
+  ml_pipeline_switch_h switchhandle;
+
+  status = ml_pipeline_switch_get_handle (handle, "ins", &type, &switchhandle);
+```
+
+Now, the switch is controllable with getting handle `ml_pipeline_switch_h`.
+
+```bash
+  ml_pipeline_h handle;
+  ml_pipeline_switch_h switchhandle;
+  ml_pipeline_switch_e type;
+
+  status = ml_pipeline_construct (pipeline, NULL, NULL, &handle);
+  status = ml_pipeline_switch_get_handle (handle, "ins", &type, &switchhandle);
+```
+
+Before destroy the pipeline, `ml_pipeline_switch_h` has to be released.
+
+```bash
+  status = ml_pipeline_switch_release_handle (switchhandle);
+```
+
+Switch can be located at the end of the pipeline like below image.
+
+![output-selector](./media/output-selector.png)
+
+```bash
+  gchar * pipeline = g_strdup ("videotestsrc is-live=true ! videoconvert ! tensor_converter ! output-selector name=outs "
+      "outs.src_0 ! tensor_sink name=sink0 async=false "
+      "outs.src_1 ! tensor_sink name=sink1 async=false");
+```
+
+### Pipeline States <a name="pipelinestate"></a>
+You can get the information about the states at [Gstreamer guide](https://gstreamer.freedesktop.org/documentation/plugin-development/basics/states.html).
+
+## Tensors Info <a name="tensorsinfo"></a>
+
+This is a structure which contains the information of the Tensors.
+`ml_tensors_info_s` is the set of tensors and `ml_tensor_info_s` is for a tensor in `ml_tensors_info_s`.
```

Comment:
plz update this section. ml_tensors_info_s is private and not opened.

Reply:
the exposed private structure was fixed.